File: backend/src/models/Order.go
package models

import "time"

type Order struct {
    OrderID     string      `db:"order_id"`
    BuyerID     string      `db:"buyer_id"`
    TotalPrice  float64     `db:"price_total"`
    Location    string      `db:"location"`
    Status      string      `db:"status"`
    ETA         time.Time   `db:"ETA"`
    CreatedAt   time.Time   `db:"created_at"`   // needed for cursor
    OrderItems  []OrderItem
}

type OrderItem struct {
    OrderItemID string  `db:"order_item_id"`
    OrderID     string  `db:"order_id"`
    ProductID   string  `db:"product_id"`   // UUID, matches Product
    Quantity    int     `db:"quantity"`
    Price       float64 `db:"price"`        // snapshot of price at time of order
}

File: frontend/src/lib/models/Order.ts
export interface OrderModel {
    orderID: string;
    buyerID: string;
    buyerEmail: string;
    totalPrice: number;
    location: string;
    status: string;
    createdAt: string;
    orderItems: OrderItemModel[];
}

export interface OrderItemModel {
    orderItemID: string;
    orderID: string;
    productID: string;
    quantity: number;
    price: number;
}

export function mapOrderResponse(raw: Record<string, unknown>): OrderModel {
    return {
        orderID: raw.orderID as string || '',
        buyerID: raw.buyerID as string || '',
        buyerEmail: raw.buyerEmail as string || '',
        totalPrice: raw.totalPrice as number || 0,
        location: raw.location as string || '',
        status: raw.status as string || '',
        createdAt: raw.createdAt as string || '',
        orderItems: (raw.orderItems as Record<string, unknown>[] || []).map(item => ({
            orderItemID: item.orderItemID as string || '',
            orderID: item.orderID as string || '',
            productID: item.productID as string || '',
            quantity: item.quantity as number || 0,
            price: item.price as number || 0
        }))
    }
}